src/rgw/rgw_http_client.cc

```diff
@@ -1165,14 +1165,14 @@ void *RGWHTTPManager::reqs_thread_entry()
 
   RWLock::WLocker rl(reqs_lock);
   for (auto r : unregistered_reqs) {
-    _finish_request(r, -ECANCELED);
+    _unlink_request(r);
```

Comment:
i'm having trouble reproducing this case in testing, but i was able to successfully exercise the all_reqs case